modules/runner/src/main/java/org/apache/ignite/internal/app/LifecycleManager.java

```diff
@@ -105,6 +110,20 @@ void onStartComplete() throws NodeStoppingException {
         }
     }
 
+    /**
+     * Represents future that will be completed when all components start futures will be completed.
+     *
+     * @return Future that will be completed when all components start futures will be completed.
+     */
+    synchronized CompletableFuture<Void> allComponentsStartFuture() {
```

Comment:
So, every call for this method will produce new future with new "clear" flow. Could you add a comment at least, that this method is planned to call only once by design?

Reply:
done

Reply:
It was proposed by @sashapolo and I agree with him, because it is a potential memory leak

Reply:
It was proposed by me, because we don't need these futures as soon as they are complete. What's the point of keeping them in memory for the whole node lifetime?

Reply:
In order to check whether component is started. I assume that when we will add true async logic we will have code similar to
aComponent.start() {
  bComponent.ready().thenRun(b -> b.fooBar());
}

Reply:
@sanpwc I didn't fully understand how this aggregated future relates to the async logic from the components. We can check that component is started using the API that you proposed for `IgniteComponents`

Reply:
component#ready or component#isStarted future will be used at any point of time, thus we will had to preserve them.
That means that we may clear only aggregated future. Slight optimization vs code simplicity, does it worth it?